test/values.test.js

```diff
@@ -501,12 +538,178 @@ describe('Values', () => {
     // versioned queries/lambdas
     assertPrint(
       new Query({ api_version: '2.12', lambda: 'X', expr: { var: 'X' } }),
-      `Query(Lambda("X", Var("X")))`
+      'Query(Lambda("X", Var("X")))'
     )
 
     assertPrint(
       new Query({ api_version: '3', lambda: 'X', expr: { var: 'X' } }),
-      `Query(Lambda("X", Var("X")))`
+      'Query(Lambda("X", Var("X")))'
+    )
+
+    assertPrint(
+      new Query({ lambda: 'X', expr: { var: 'X' }, api_version: '3' }),
+      'Query(Lambda("X", Var("X")))'
+    )
+  })
+
+  test('pretty print despite argument order', () => {
+    // Match
+    assertPrint(
+      new Query({
+        api_version: '3',
+        expr: {
+          match: {
+            index: 'idx',
+          },
+          terms: ['str', 10],
+        },
+        lambda: '_',
+      }),
+      'Query(Lambda("_", Match(Index("idx"), ["str", 10])))'
+    )
+
+    assertPrint(
+      new Query({
+        expr: {
+          terms: 10,
+          match: {
+            index: 'idx',
+          },
+        },
+        api_version: '3',
+        lambda: '_',
+      }),
+      'Query(Lambda("_", Match(Index("idx"), 10)))'
+    )
+
+    // Filter an array
+    assertPrint(
+      new Query({
+        collection: [1, 2, 3],
+        api_version: '3',
```

Comment:
Filter expressions doesn't have `api_version` argument.

Reply:
Good call. I'll remove this 👍